packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+        columnId="lastName"
+    ></nimble-text-field-column>
+    <nimble-text-field-column
+        slot="columns"
+        columnId="profession"
+    ></nimble-text-field-column>
+</nimble-table>
+```
+
+_Placeholder for other design details as they are created_
+
+### API
+
+_Component Name_
+
+`nimble-table`
+
+_Feature APIs_
+
+The various APIs/features of the `nimble-table` will be split up amongst several different HLD documents. This section will serve to list them out and link to them as they become available:
+
+-   Data API :
+    -   Define what the interface is for setting/getting data on the table
+    -   Illustrate how the data is hooked up to the TanStack API
+    -   How is data hierarchy provided and represented in the DOM
+-   Column Definitions
+    -   Define the interface we will provide for the column providers/components (i.e., width, sorting, allowSort, allowGrouping, etc...)
+    -   Define the base implementation for the column providers that other column providers can extend.
+    -   List the set of column providers that Nimble will provide and provide their respective APIs where unique (e.g., formatter for DateTime column)
+    -   Define the table-level APIs for setting the columns
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type?
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
+-   Grouping
+    -   Define any interactive mechanism, if any, to provide grouping
+    -   Define table-level API for setting grouping
+    -   Define events raised when grouping changes
+-   Sorting
+    -   Define the table-level API for setting sorting
+    -   Define events raised when sorting changes
+-   Action Menu
+    -   Define how the action menu gets associated with a particular column
+    -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
+
+_Properties_
+
+Placeholder
+
+_Events_
+
+Placeholder
+
+### Anatomy
+
+_Slots_
+
+-   `slottedColumns` (Placeholder for column slots)
+-   `actionMenu` (Placeholder for action menu)
+
+### Angular integration
+
+Angular support should be accomplished through the typical directive patterns.
+
+### Blazor integration
+
+Blazor support should be accomplished through the typical integration patterns. It should be noted that if the Nimble `Table` component is generic, than the Blazor component will mirror that generic API.
+
+---
+
+## Implementation
```

Comment:
One outcome we'd like to drive towards is a clear articulation of next steps after this spec is approved. 
1. what implementation can be started?
2. what HLDs do we write first?
3. which parts can be done in parallel and what are the dependencies?
4. how to we track all of that in GitHub issues? Are the tasks in Nate's table GitHub project useful or should we start fresh with dev-focused taskes?

Do you think that will fall naturally out of this spec or will it require a separate effort to coordinate? If the latter, feel free to drive that yourself or try to convince someone else to do it.

Reply:
As discussed elsewhere,  we will create an issue that represents the first chunk of work that meets a particular milestone.